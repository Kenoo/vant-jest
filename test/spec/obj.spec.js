// __tests__/example.test.js
import { resolve } from 'q';
import example from '../obj';

describe('test object data is right', ()=> {

    test('square fn', () => {
        const mockFn = jest.fn(example.square);
        mockFn('3', 4);
        expect(mockFn).toHaveBeenCalled();

        expect(example.square('2', 3)).toBe(6);
        expect(example.square(0, '3b')).toBe(NaN);
        expect(example.square('5c', '3b')).toBe(NaN);
        expect(example.square(5, 4)).toBe(20);

    });

    test('asyncSquare fn', ()=> {
        let fnPromise = (a)=>{
            return new Promise((resolve, reject)=>{
                if (typeof a != 'number') reject(new Error('参数必须是number类型'));

                setTimeout(()=>{resolve(a)}, 1000);
            });
        }
        expect(example.asyncSquare(fnPromise(2), 3)).resolves.toBe(6);
    }, 2000);

    test('sum fn', () => {
        expect(example.sum(2, 3)).toBe(5);
        expect(example.sum(0.1, 0.2)).toBeCloseTo(0.3);
    });

    test('onPress gets called with the right thing', () => {
       
    });

    test('obj toMatchSnapshot', ()=> {
        expect(example).toMatchSnapshot();
    });
});



