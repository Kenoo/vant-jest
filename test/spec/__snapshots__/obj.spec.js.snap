// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test object data is right obj toMatchSnapshot 1`] = `
Object {
  "array": Array [
    1,
    2,
    3,
  ],
  "asyncSquare": [Function],
  "boolean": true,
  "class": Bar {
    "array": Array [
      1,
      2,
      3,
    ],
  },
  "number": 1234,
  "object": Object {
    "bar": Object {
      "buzz": Array [
        1,
        2,
        3,
      ],
      "fiz": 1,
    },
    "baz": "test",
  },
  "square": [Function],
  "string": "baz",
  "sum": [Function],
  "symbol": Symbol(a.b.c),
}
`;
